impeller/display_list/paint.cc

```diff
@@ -34,6 +35,22 @@ using DlRect = flutter::DlRect;
 using DlIRect = flutter::DlIRect;
 using DlPath = flutter::DlPath;
 
+// static
+bool Paint::CanApplyOpacityPeephole(const Paint& paint) {
+  if (paint.blend_mode != BlendMode::kSourceOver) {
+    return false;
+  }
+  if (paint.invert_colors || paint.mask_blur_descriptor.has_value() ||
```

Comment:
Also, SaveLayer shouldn't apply mask blur etiher.

Only alpha of the color, IF, CF and blend.